randomGenerator/app.js

```diff
@@ -0,0 +1,66 @@
+/*
+Optional bonus work on conditionals and functions:
+
+    Write a script that can generate random addresses
+    As a first step, create arrays that contain dummy data for each of the following: street number, street name, city name, state name, and zip code
+    Your script should randomly select one item from each of these arrays and then use them to construct a random address
+    Each time you load or reload the page in a browser, your program should print a new randomly-generated address to the terminal. For example:
+    
+    34578 Dolphin Street, Wonka NY, 44506
+*/
+
+//below using function inside of function
+function randomNumberGenerator() {
+	return (Math.floor(((Math.random() * 9) + 1)));
```

Comment:
Por que você acrescentou 1 depois de gerar um número aleatório?
Dessa forma, seu código nunca vai gerar `0`, então todos os primeiros elementos dos seus arrays serão inutilizados.

Math.random() gera um número de ponto flutuante entre 0 (inclusivo) e 1 (exclusivo), ou seja, um número na faixa:
`0.0000 .. 0.9999999999999999`

Removendo o 1 adicionado e multiplicando os extremos por 9, temos:
`0 .. 8.99999999999999999.8`
mantendo seu `Math.floor` nesse caso, o número 9 nunca será atingido.

Agora, se ao invés você usar `Math.ceil` (arredondar pra cima) o 9 vai ser atingido, e o 0 também (eventualmente), mas a chance de sair o 0 é muito menor do que a dos outros números, porque ele só vai sair quando o resultado de `Math.random()` for exatamente 0. (o que é melhor que o seu jeito de implementar, mas não tão melhor assim)

Se você quiser um distribuição mais justa mantendo sua lógica do random, você poderia fazer:
`Math.round(Math.random() * 9)`
porém ainda não é o melhor caso, pois você teria os seguintes intervalos:
```
  0 ~ 0.499999... -> 0
0.5 ~ 1.499999... -> 1
1.5 ~ 2.499999... -> 2
...
8.5 ~8.999999... -> 9
```

Ou seja, a faixa de valores que mapeiam para 0 e 9 tem metade das chances de serem atingidas, o que torna a distribuição não-linear.

Para resolver esse problema, basta você usar 10 ao invés de 9 e manter seu `Math.floor`:
```js 
Math.floor(Math.random() * 10);
//ou de forma mais genérica
Math.floor(Math.random() * (maxValue + 1));
```
Isso da certo porque a função `Math.random()` foi feita exatamente com isso em mente. E é exatamente por isso que a função não inclui o número 1 dentro da faixa de valores possíveis gerados. Dessa forma você tem um distribuição linear que agora só depende da implementação de `Math.random()` e não do seu código.